src/google/protobuf/compiler/objectivec/objectivec_file.cc

```diff
@@ -115,6 +115,9 @@ void FileGenerator::GenerateHeader(io::Printer *printer) {
   printer->Print(
       "// @@protoc_insertion_point(imports)\n"
       "\n"
+      "#pragma clang diagnostic push\n"
+      "#pragma clang diagnostic ignored \"-Wdeprecated-declarations\"\n"
+      "\n"
```

Comment:
This warning suppression should not affect building at all.
I had to add "Wdeprecated-declarations" warnings suppression because in all places where deprecated message/field/enum is used warning was generated. Suppression starts at file begin and end at file end. In *.m files such usages are in parsing code, in *.h files - in places where some message uses another deprecated message as a field.